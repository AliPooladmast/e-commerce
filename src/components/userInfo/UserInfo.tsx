import React from "react";
import style from "./userInfo.module.scss";
import {
  CalendarToday,
  LocationSearching,
  MailOutline,
  PermIdentity,
  PhoneAndroid,
} from "@mui/icons-material";

interface IUser {
  username: string;
  _id: string;
  isAdmin: boolean;
  token: string;
  createdAt: string;
  fullname?: string;
  email?: string;
  phone?: string;
  address?: string;
  img?: string;
  title?: string;
}

const UserInfo = ({ user }: { user: IUser }) => {
  return (
    <div className={style.UserInfo}>
      <div className={style.Top}>
        <img
          src={user.img || "assets/icons/no-avatar.svg"}
          alt="user profile"
        />
        <div className={style.TitleContainer}>
          <span className={style.Name}>{user.username}</span>
          <span className={style.Title}>{user.title}</span>
        </div>
      </div>
      <div className={style.Bottom}>
        <span className={style.Title}>Account Details</span>
        <div className={style.Info}>
          <PermIdentity className={style.Icon} />
          <span>{user.fullname || user.username}</span>
        </div>
        <div className={style.Info}>
          <CalendarToday className={style.Icon} />
          <span>{user.createdAt?.split("T")?.[0] || "-- / -- / ----"}</span>
        </div>
        <span className={style.Title}>Contact Details</span>
        <div className={style.Info}>
          <PhoneAndroid className={style.Icon} />
          <span>{user.phone || "-------"}</span>
        </div>
        <div className={style.Info}>
          <MailOutline className={style.Icon} />
          <span>{user.email}</span>
        </div>
        <div className={style.Info}>
          <LocationSearching className={style.Icon} />
          <span>{user.address || "-----"}</span>
        </div>
      </div>
    </div>
  );
};

export default UserInfo;
