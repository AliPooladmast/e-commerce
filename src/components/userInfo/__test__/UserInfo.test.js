const { render } = require("@testing-library/react");
const { BrowserRouter } = require("react-router-dom");
import UserInfo from "../UserInfo";
import renderer from "react-test-renderer";

const user = {
  username: "username",
  _id: "id",
  isAdmin: "isAdmin",
  token: "token",
  createdAt: "createdAt",
  fullname: "fullname",
  email: "email",
  phone: "phone",
  address: "address",
  img: "img",
  title: "title",
};

it("match the snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <UserInfo user={user} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
