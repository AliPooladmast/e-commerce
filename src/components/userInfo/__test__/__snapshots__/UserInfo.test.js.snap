// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match the snapshot 1`] = `
<div
  className="UserInfo"
>
  <div
    className="Top"
  >
    <img
      alt="user profile"
      src="img"
    />
    <div
      className="TitleContainer"
    >
      <span
        className="Name"
      >
        username
      </span>
      <span
        className="Title"
      >
        title
      </span>
    </div>
  </div>
  <div
    className="Bottom"
  >
    <span
      className="Title"
    >
      Account Details
    </span>
    <div
      className="Info"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icon css-i4bv87-MuiSvgIcon-root"
        data-testid="PermIdentityIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <span>
        fullname
      </span>
    </div>
    <div
      className="Info"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icon css-i4bv87-MuiSvgIcon-root"
        data-testid="CalendarTodayIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
        />
      </svg>
      <span>
        createdAt
      </span>
    </div>
    <span
      className="Title"
    >
      Contact Details
    </span>
    <div
      className="Info"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icon css-i4bv87-MuiSvgIcon-root"
        data-testid="PhoneAndroidIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
        />
      </svg>
      <span>
        phone
      </span>
    </div>
    <div
      className="Info"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icon css-i4bv87-MuiSvgIcon-root"
        data-testid="MailOutlineIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
        />
      </svg>
      <span>
        email
      </span>
    </div>
    <div
      className="Info"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium Icon css-i4bv87-MuiSvgIcon-root"
        data-testid="LocationSearchingIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.94 11c-.46-4.17-3.77-7.48-7.94-7.94V1h-2v2.06C6.83 3.52 3.52 6.83 3.06 11H1v2h2.06c.46 4.17 3.77 7.48 7.94 7.94V23h2v-2.06c4.17-.46 7.48-3.77 7.94-7.94H23v-2h-2.06zM12 19c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
        />
      </svg>
      <span>
        address
      </span>
    </div>
  </div>
</div>
`;
